### PR TITLE
Adding x-www-form-urlencoded

### PR DESCRIPTION
**Pull request by anderskvist**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [X] I have read and understand the pull request rules.

# Description

Adding x-www-form-urlencoded to allow monitors to login via webforms.
Resolves https://github.com/louislam/uptime-kuma/issues/1351
Resolves https://github.com/louislam/uptime-kuma/issues/1332

Depends on https://github.com/louislam/uptime-kuma/pull/3498 as that fixes issue with anything but HTTP monitors that tries to use XML or x-www-form-urlencoded.

## Type of change

Please delete any options that are not relevant.

- New feature (non-breaking change which adds functionality)

## Checklist

- [X] My code follows the style guidelines of this project
- [ ] I ran ESLint and other linters for modified files
- [X] I have performed a self-review of my own code and tested it
- [ ] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [X] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

Please do not use any external image service. Instead, just paste in or drag and drop the image here, and it will be uploaded automatically.

### CODE REVIEW

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
I think this is missing a part of the frontend:

This is the UI of f.ex. Postman:
image

This PR looks like this:
image

I think you are also missing the part where this is urlencoding the string (which differentiates `form-data` from `x-www-form-urlencoded`)

**Comment by anderskvist** [comment]
@CommanderStorm Didn't see your message, but I actually just added this example :D

File: server/model/monitor.js [hunk 1]
```diff
@@ -359,6 +359,9 @@ class Monitor extends BeanModel {
                             } catch (e) {
                                 throw new Error("Your JSON body is invalid. " + e.message);
                             }
+                        } else if (this.httpBodyEncoding === "form") {
+                            bodyValue = this.body;
+                            contentType = "application/x-www-form-urlencoded; charset=utf-8";
```
Comment (CommanderStorm) [inline server/model/monitor.js:364]: See https://stackoverflow.com/questions/16819502/application-x-www-form-urlencoded-and-charset-utf-8
=> not needed

```suggestion
                            contentType = "application/x-www-form-urlencoded";
```

File: src/pages/EditMonitor.vue [hunk 2]
```diff
@@ -930,6 +931,9 @@ message HealthCheckResponse {
   </soap:Body>
 </soap:Envelope>` ]);
             }
+            if (this.monitor && this.monitor.httpBodyEncoding && this.monitor.httpBodyEncoding === "form") {
+                return this.$t("Example:", [ `key1=value1&key2=value2` ]);
```
Comment (CommanderStorm) [inline src/pages/EditMonitor.vue:935]: The linter does like it this way 😉 

```suggestion
                return this.$t("Example:", [ "key1=value1&key2=value2" ]);
```

**Comment by CommanderStorm** [comment]
Could you add the following to the description:
Resolves https://github.com/louislam/uptime-kuma/issues/1351
Resolves https://github.com/louislam/uptime-kuma/issues/1332

File: src/pages/EditMonitor.vue [hunk 3]
```diff
@@ -930,6 +931,9 @@ message HealthCheckResponse {
   </soap:Body>
 </soap:Envelope>` ]);
             }
+            if (this.monitor && this.monitor.httpBodyEncoding && this.monitor.httpBodyEncoding === "form") {
```
Comment (Computroniks) [inline src/pages/EditMonitor.vue:934]:
```suggestion
            if (this.monitor && this.monitor.httpBodyEncoding === "form") {
```

Does the middle condition serve any purpose? I know that XML also has it so we might want to keep it in here for uniformity, but does it actually do anything useful?

**Review by Computroniks — COMMENTED** [review]
Nice work :+1:

**Comment by dakriy** [comment]
Since this is basically done, it is a very simple and useful change, its prerequisites are already merged, and the release hasn't happened yet. Is there any chance of moving this to the 1.23.0 release so that it can be used before 2.0 comes out? It would be very useful for me and I don't want to wait for 2.0 to start using it.

File: src/pages/EditMonitor.vue (hunk 3 above)
Comment (louislam) [reply src/pages/EditMonitor.vue:934]: I think it could be simpler.

**Comment by foliovision** [comment]
Hello @louislam 

It's too bad this is not part of 1.23.2 release.

Any idea when version 2.0.0 will be out? Is this really a big change that could not go to 1.23.3?

Please let us know.

Thanks,
Martin

**Comment by CommanderStorm** [comment]
@foliovision Please refrain from unnessesary pings

> Any idea when version 2.0.0 will be out

As any sane open source repository we don't work under deadlines.
See the [tracking milestone](https://github.com/louislam/uptime-kuma/milestone/24) for the progress.
See [#NoEsitmates](https://www.youtube.com/watch?v=QVBlnCTu9Ms) why asking for esimates is a bad idea.

> Is this really a big change that could not go to 1.23.3?

the v1 series will only recieve bug fixes as by our policy laied out here: 
https://github.com/louislam/uptime-kuma/releases/tag/1.23.0

Given that this is a feature, this will be inluded in the v2 release and no special release will be drafted for this.

**Comment by robertomurta** [comment]
Was this issue of sending as formdata accepted before the deletion of the branch?

**Pull request merged**